extensions/notebook/src/book/bookModel.ts

```diff
@@ -85,8 +89,8 @@ export class BookModel implements azdata.nb.NavigationProvider {
 				);
 				this._bookItems.push(book);
 			} catch (e) {
-				let error = e instanceof Error ? e.message : e;
-				vscode.window.showErrorMessage(error);
+				this._errorMessage = e instanceof Error ? e.message : e;
```

Comment:
Instead of doing it this way you should use https://github.com/microsoft/azuredatastudio/blob/master/extensions/notebook/src/common/apiWrapper.ts#L44 and then mock the ApiWrapper for tests so you can capture when showErrorMessage is called.